### PR TITLE
Use subtests. Need to stob building on go < 1.7.

### PR DESCRIPTION
**Pull request by thedodd**
### note
This PR is based on google/jsonapi#38. Once it is merged into master, this PR will be rebased. There is only **one** commit unique to this PR.

### CODE REVIEW

**Comment by thedodd** [comment]
After #38 is merged, I will rebase this one and it should resolve any conflicts.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 language: go
 go:
-  - 1.6.x
```
Comment (thedodd) [inline .travis.yml]: Subtests were introduced in `1.7`, so we can not use 1.6 in tests now.

Comment (aren55555) [inline .travis.yml]: I think 1.6 should be supported until 1.8 is released (any day now).

Comment (thedodd) [inline .travis.yml]: 1.8 has dropped. We should be good to go on this now.

**Pull request merged**